addons/sale_crm/models/sale_order.py

```diff
@@ -12,4 +12,7 @@ class SaleOrder(models.Model):
         domain="[('type', '=', 'opportunity'), '|', ('company_id', '=', False), ('company_id', '=', company_id)]")
 
     def action_confirm(self):
-        return super(SaleOrder, self.with_context({k:v for k,v in self._context.items() if k != 'default_tag_ids'})).action_confirm()
+        res = super(SaleOrder, self.with_context({k: v for k, v in self._context.items() if k != 'default_tag_ids'})).action_confirm()
+        for order in self:
+            self.opportunity_id._update_revenues_from_so(order)
```

Comment:
Seems we could have a m2m-like behavior we don't want 
```suggestion
        for order in self:
            order.opportunity_id._update_revenues_from_so(order)
```